### PR TITLE
[TASK] Add information about DI and Caches

### PR DESCRIPTION
**Pull request by linawolf**
releases: main, 12.4

### CODE REVIEW

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst [hunk 1]
```diff
@@ -89,6 +92,25 @@ Method :php:`getPrerequisites()`
             ];
         }
 
+After creating the new upgrade wizard, delete all caches in  
+:guilabel:`Admin tools > Maintanance > Flush TYPO3 and PHP Cache` or via console
+command:
+
+.. tabs::
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst:99]: We have an include you can use.

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst [hunk 2]
```diff
@@ -42,6 +42,9 @@ The registration of an upgrade wizard is done directly in the class by adding
 the class attribute :php:`\TYPO3\CMS\Install\Attribute\UpgradeWizard`. The
 :ref:`unique identifier <upgrade-wizards-identifier>` is passed as an argument.
 
+Your extension must have a :ref:`Services.yaml <t3coreapi:extension-configuration-services-yaml>`
+with `autowire = true` for the upgrade wizard to be detected.
```
Comment (brotkrueml) [inline Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst:46]: Actually, it *should* also work with autowire=false and tagging manually in Services.yaml:

    services:
       YourVender\YourExtension\UpgradeWizard\YourUpgradeWizard:
          tags:
             - name: install.upgradewizard
               identifier: your-upgrade-wizard

Comment (brotkrueml) [reply Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst:46]: And: Do you mean autowire or autoconfigure? I think, this should be autoconfigure=true without tagging manually.

Comment (brotkrueml) [reply Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst:46]: Yup, should be autoconfigure.

Autowiring:
"... reads the type-hints on your constructor (or other methods) and automatically passes the correct services to each method."
https://symfony.com/doc/current/service_container/autowiring.html

Comment (brotkrueml) [reply Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst:46]: So, wie should add something like this:
https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/Backend/ContextualMenu.html#step-3-registration

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst (hunk 1 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst:99]: Where do we have that?

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst (hunk 2 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst:46]: @brotkrueml would you like to take over? It seems like I am not deep enough into this, just noted that without a services.yaml it deosnt work

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst (hunk 1 above)
Comment (brotkrueml) [reply Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst:99]: https://github.com/TYPO3-Documentation/TYPO3CMS-Reference-CoreApi/blob/main/Documentation/_includes/CliCacheFlush.rst.txt

Comment (brotkrueml) [reply Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst:99]: Done

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst (hunk 2 above)
Comment (brotkrueml) [reply Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst]: Done

File: Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst (hunk 1 above)
Comment (linawolf) [reply Documentation/ExtensionArchitecture/HowTo/UpdateExtensions/UpdateWizards/Creation.rst]: Thank you!

**Review by garvinhicking — APPROVED** [review]
Thanks for working this out, looks good to me! :)

**Pull request merged**